examples/password_strength/src/text_input.rs

```diff
@@ -0,0 +1,45 @@
+use yew::prelude::*;
+
+#[derive(Properties, Clone)]
+pub struct Props {
+    pub value: String,
+    pub onchange: Callback<String>,
+}
+
+pub struct Model {
+    props: Props,
+    link: ComponentLink<Self>,
+}
+
+pub type TextInput = Model;
```

Comment:
Same as above.
Oddly, I kind of like this pattern but it definitely isn't consistent with all the other examples and I fear it might be a tad confusing to new users (especially those new to Rust).

Reply:
I would actually suggest the reverse; Yew was one of the first (if not the first) libraries I looked at when learning Rust and I always found it confusing that the model had the component trait implemented on it without acknowledging that the model is the component itself. The examples generally aren't very consistent anyways, wouldn't it be good to cultivate a variety of approaches?

Reply:
> wouldn't it be good to cultivate a variety of approaches?

Hmm, I can't say I disagree. I don't personally like this style because it adds even more boilerplate (although it's only a single line) and kind of fights against Yew principles because it tries to separate something Yew normally doesn't. That being said, I think it's valuable to keep this for users who are familiar with the MVC model who would expect the model to be separate from the controller.

> The examples generally aren't very consistent anyways

I put quite a lot of effort into making the examples align as closely as possible to a common guideline.
Doesn't mean that I've necessarily achieved that goal, but I think it would be a good idea to add an explanation for this pattern (either to the README or somewhere in the code).

Reply:
> I put quite a lot of effort into making the examples align as closely as possible to a common guideline 

Woops, yeah my knowledge of this may be out of date now since the last time I looked at them all was a few months ago. I agree, let's add at least an explanation to the README. Will update soon.